planning.py

```diff
@@ -1,11 +1,57 @@
 """Planning (Chapters 10-11)
 """
 
+import copy
 import itertools
 from search import Node
-from utils import Expr, expr, first, FIFOQueue
+from utils import Expr, expr, first, FIFOQueue, partition
 from logic import FolKB
+from logic import fol_bc_and
+
+
+class PlanningKB:
+    """ A PlanningKB contains a set of Expr objects that are immutable and hashable.
+     With its goal clauses and its accompanying h function, the KB
+     can be used by the A* algorithm in its search Nodes. (search.py) """
+    def __init__(self, goals, initial_clauses=None):
+        if initial_clauses is None:
+            initial_clauses = []
+        self.goal_clauses = frozenset(goals)
+        self.clause_set = frozenset(initial_clauses)
+
+    def __eq__(self, other):
+        """search.Node has a __eq__ method for each state, so this method must be implemented too."""
```

Comment:
I would like to see some explanation as to what the function does, so that people don't have to jump to the `search` module to find out.